Add test coverage for query endpoints and config

aaron-plahn opened the pull request.

yaanahuu2 commented on apps/api/src/domain/domainModelValidators/domainModelValidators.spec.ts:
```diff
@@ -49,6 +49,7 @@ const validVocabularyListDTO: PartialDTO<VocabularyList> = {
 const validTagDTO: PartialDTO<Tag> = {
```
Test publication filtering

aaron-plahn replied:
I added a story for this [here](https://www.pivotaltracker.com/story/show/181563197).

yaanahuu2 commented on apps/api/src/domain/domainModelValidators/tagValidator.ts:
```diff
@@ -20,11 +21,15 @@ const tagValidator: DomainModelValidator = (
   if (!isStringWithNonzeroLength(text))
     innerErrors.push(new TagHasNoTextError(id));
 
+  // TODO Validate inherited properties on the base class
```
Do this now

aaron-plahn replied:
On second thought, I'm going to wait. There's an important design decision to be made here and I don't want to go there prematurely.

yaanahuu2 commented on apps/api/src/domain/domainModelValidators/vocabularyListValidator.ts:
```diff
@@ -5,6 +5,7 @@ import { PartialDTO } from '../../types/partial-dto';
 import { VocabularyList } from '../models/vocabulary-list/entities/vocabulary-list.entity';
```
Remove old validation pattern on entity classes

aaron-plahn replied:
Done

yaanahuu2 commented on testData.json:
```diff
@@ -1 +1 @@
-{"tags":[{"_key":"0","published":true,"type":"tag","text":"plants"},{"_key":"1","published":true,"type":"tag","text":"animals"},{"_key":"2","published":true,"type":"tag","text":"placenames"},{"_key":"3","published":true,"type":"tag","text":"songs"},{"_key":"4","published":true,"type":"tag","text":"legends"}],"terms":[{"_key":"1","published":false,"type":"term","term":"Chil-term-1","termEnglish":" ...
```
Gitignore this file

aaron-plahn replied:
Done. I moved the write location to the same directory of the test as well.

aaron-plahn commented on apps/api/src/app/app.controller.ts:
```diff
@@ -14,7 +14,6 @@ export class AppController {
   @UseGuards(AuthGuard('jwt'))
   @Get('hello')
   getData(): MediaItem {
-    console.log(`You called get!`);
```
Done.

aaron-plahn commented on apps/api/src/app/config/buildConfigFilePath.ts:
```diff
@@ -0,0 +1,16 @@
+    throw new InternalError(`Invalid .env file path: ${path}`);
+  }
+
+  console.log({
```
Done.

aaron-plahn commented on apps/api/src/app/config/config.spec.ts:
```diff
@@ -0,0 +1,114 @@
+ * guidance in how to setup more complex e2e tests of our api.
+ */
+
+const bootstrapApp = (configFileName: string) =>
```
Done.

aaron-plahn commented on apps/api/src/app/config/config.spec.ts:
```diff
@@ -0,0 +1,114 @@
+
+  describe('when all environment variables are valid', () => {
+    beforeAll(async () => {
+      const moduleRef = await bootstrapApp('sample');
```
Apparently we do. the .compile method returns a promise.

aaron-plahn commented on apps/api/src/app/config/env.validation.ts:
```diff
@@ -0,0 +1,99 @@
+  AUTH0_AUDIENCE: string;
+}
+
+export const allEnvironmentVariableKeys = [
```
Done.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/appBaseRoute.e2e.spec.ts:
```diff
@@ -0,0 +1,82 @@
+    message: 'Welcome to the COSCRAD API!',
+  };
+
+  describe('the environment', () => {
```
Done.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+    jest.resetModules();
+
+    process.env = {
+      ...originalEnv,
```
Done

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+      imports: [
+        ConfigModule.forRoot({
+          isGlobal: true,
+          envFilePath: `${process.cwd()}/apps/api/src/app/config/${
```
Done.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+    await app.init();
+
+    /**
+     * We empty the collections before **and** after each test, just in case.
```
Done.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+import { Test } from '@nestjs/testing';
+import { EntityType, entityTypes } from 'apps/api/src/domain/types/entityType';
+import { ArangoConnectionProvider } from 'apps/api/src/persistence/database/arango-connection.provider';
+import TestRepositoryProvider from 'apps/api/src/persistence/repositories/TestRepositoryProvider';
```
I'll wait until we get the story for that.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+      await app.close();
+    });
+  });
+});
```
I added a story for this [here](https://www.pivotaltracker.com/story/show/181561510)

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entityDescriptions.e2e.spec.ts:
```diff
@@ -0,0 +1,46 @@
+      imports: [
+        ConfigModule.forRoot({
+          isGlobal: true,
+          envFilePath: `${process.cwd()}/apps/api/src/app/config/${
```
Done.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+import { INestApplication } from '@nestjs/common';
```
Done.

aaron-plahn commented on apps/api/src/app/controllers/__tests__/entities.e2e.spec.ts:
```diff
@@ -0,0 +1,136 @@
+
+const originalEnv = process.env;
+
+const entityType = entityTypes.term;
```
Done.

aaron-plahn commented:
In some cases I decided to postpone or create a story for the work suggested in comments. In the former case, I've left the corresponding issue marked 'unresolved'. Otherwise, I've addressed all feedback.
